orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/support/TargetServerGroupResolver.groovy

```diff
@@ -106,35 +106,30 @@ class TargetServerGroupResolver {
    */
   static TargetServerGroup fromPreviousStage(Stage stage) {
     // The DetermineTargetServerGroupStage has all the TargetServerGroups we want - go find it!
-    def dtsgStage = stage.execution.stages.find {
-      isDTSGStage(it) && stage.ancestors().contains(it)
-    }
+    def dtsgStage = stage.ancestors().find { isDTSGStage(it) }
```

Comment:
This is the crucial change. Without this if B depends on A and both are doing `DetermineTargetServerGroups` we'd always pick A's.